src/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore/Internal/DiagnosticsEntityFrameworkCoreLoggerExtensions.cs

```diff
@@ -8,127 +8,111 @@ namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Internal
+            LogLevel.Error,
+            new EventId(1, "NoContextType"),
+            "No context type was specified. Ensure the form data from the request includes a contextTypeName value, specifying the context to apply migrations for.");
+
+        private static readonly Action<ILogger, string, Exception> _invalidContextType = LoggerMessage.Define<string>(
+            LogLevel.Error,
+            new EventId(2, "InvalidContextType"),
+            "The context type '{ContextTypeName}' could not be loaded. Ensure this is the correct type name for the context you are trying to apply migrations for.");
+
+        private static readonly Action<ILogger, string, Exception> _contextNotRegistered = LoggerMessage.Define<string>(
+            LogLevel.Error,
+            new EventId(3, "ContextNotRegistered"),
+            "The context type '{ContextTypeName}' was not found in services. This usually means the context was not registered in services during startup. You probably want to call AddScoped<>() inside the UseServices(...) call in your application startup code.");
+
+        private static readonly Action<ILogger, string, Exception> _requestPathMatched = LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(4, "RequestPathMatched"),
+            "Request path matched the path configured for this migrations endpoint({RequestPath}). Attempting to process the migrations request.");
+
+        private static readonly Action<ILogger, string, Exception> _applyingMigrations = LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(5, "ApplyingMigrations"),
+            "Request is valid, applying migrations for context '{ContextTypeName}'");
+
+        private static readonly Action<ILogger, string, Exception> _migrationsApplied = LoggerMessage.Define<string>(
+            LogLevel.Debug,
+            new EventId(6, "MigrationsApplied"),
+            "Migrations successfully applied for context '{ContextTypeName}'.");
+
+        private static readonly Action<ILogger, string, Exception> _migrationsEndPointMiddlewareException = LoggerMessage.Define<string>(
+            LogLevel.Error,
+            new EventId(7, "Exception"),
+            "An error occurred while applying the migrations for '{ContextTypeName}'. See InnerException for details:");
+
+        // DatabaseErrorPageMiddleware
+        private static readonly Action<ILogger, Type, Exception> _attemptingToMatchException = LoggerMessage.Define<Type>(
+            LogLevel.Debug,
+            new EventId(1, "AttemptingToMatchException"),
+            "{ExceptionType} occurred, checking if Entity Framework recorded this exception as resulting from a failed database operation.");
+
+        private static readonly Action<ILogger, Exception> _noRecordedException = LoggerMessage.Define(
+            LogLevel.Debug,
+            new EventId(2, "AttemptingToMatchException"),
```

Comment:
Different ID but same name as above?

![https://media.giphy.com/media/sCZy8qiUqFAeQ/200.gif](https://media.giphy.com/media/sCZy8qiUqFAeQ/200.gif)

Reply:
Copy paste error >.< Thank you!